### ISSUE TITLE
Website not accessible: ERR_TOO_MANY_REDIRECTS

### ISSUE THREAD

**Issue by yuchi** [liferay/clay#528]
```
$ curl -I https://claycss.com/
HTTP/2 302
date: Thu, 08 Feb 2018 10:57:55 GMT
content-length: 0
set-cookie: __cfduid=XXXXXXX; expires=Fri, 08-Feb-19 10:57:55 GMT; path=/; domain=.claycss.com; HttpOnly
cache-control: no-cache
location: https://claycss.com/
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 3e9e1220cb4e434c-MXP
```

Pay attention to where the `location` header send me.

**Comment by jbalsas** [comment]
Thanks for reporting, @yuchi, we've been tracking this down, not sure what's going on, but I'm sure @zenorocha will help us out! 🙏

**Comment by yuchi** [comment]
My bet is on wrong https proxy setup. The app is being proxied as http and doesn't know the user is using https, hence the redirect.

**Comment by zenorocha** [comment]
Problem solved: yesterday a member of Liferay's IT team noticed that all assets on claycss.com were responding with no-cache flags set. So he decided to enable the CDN feature on Cloudflare (that cloud icon). The problem is that when this configuration is set on Cloudflare, WeDeploy is unable to resolve the domain.

**Comment by yuchi** [comment]
Curious! Thanks for the info!

**Comment by zenorocha** [comment]
We investigated a little bit more and here's what we found.

The issue was that we were using Cloudflare's Flexible SSL, which means that Cloudflare will use SSL to deliver to the client, even if the server doesn't support SSL. So there was a redirect loop between the browser requesting port 443, Cloudflare hitting the server on port 80, the server telling the client to hit 443, the client hits 443 again, Cloudflare hits port 80 again, and around and around we go.

To fix it, we set it to Full (strict) SSL, which means that Cloudflare will only serve SSL traffic if WeDeploy is showing a valid and unexpired SSL cert.

In practical terms, claycss.com is now being served with CDN enabled and all assets are being cached. 

```
curl -svo /dev/null https://claycss.com/images/home/news.svg 2>&1 | grep CF-Cache-Status
```
